platform/ios/src/MGLMapView.mm

```diff
@@ -1337,6 +1337,12 @@ - (void)handlePinchGesture:(UIPinchGestureRecognizer *)pinch
 
         [self unrotateIfNeededForGesture];
     }
+    
+    if ([self.delegate respondsToSelector:@selector(mapView:shouldChangeFromCamera:toCamera:)]
+        && ![self.delegate mapView:self shouldChangeFromCamera:oldCamera toCamera:self.camera])
+    {
+        self.camera = oldCamera;
+    }
```

Comment:
This seems to happen too late in the process? i.e. `setScale()` has already been called above, versus the way the way other gestures are handled in this PR.